Use contextmanager when using compression_util.get_fileobj()

mvdbeek opened the pull request.

mvdbeek committed:
Use contextmanager when using compression_util.get_fileobj()

nsoranzo commented on lib/galaxy/datatypes/tabular.py:
```diff
@@ -788,45 +788,41 @@ def sniff( self, filename ):
+                if count > 0:
+                    return True
+        except Exception:
             pass
```
You can remove the external `try/except`, exceptions in a sniffer are already caught in `galaxy.datatypes.sniff.guess_ext()` and are equivalent to the following `return False`.

mvdbeek replied:
So then I can also remove the inner `try/except`, right?

mvdbeek committed:
Remove outer try/except in Eland sniffer

Exceptions in a sniffer are already caught in
galaxy.datatypes.sniff.guess_ext() and are equivalent to `return False`.
Thanks @nsoranzo!

mvdbeek committed:
Also remove the inner try/except

nsoranzo commented:
Thanks @mvdbeek!
